openmdao/visualization/meta_model_viewer/meta_model_visualization.py

```diff
@@ -0,0 +1,553 @@
+            else:
+                setattr(self, name, slider_object)
+                obj = getattr(self, name)
+                obj.on_change('value', self._update)
+
+        # Length of inputs and outputs
+        self.num_of_inputs = len(self.input_list)
+        self.num_of_outputs = len(self.output_list)
+
+        # Positional indicies
+        self.x_index = 0
+        self.y_index = 1
+        self.output_variable = self.output_list.index(self.output_select.value)
+
+        # Most data sources are filled with initial values
+        self.slider_source = ColumnDataSource(data=self.input_data_dict)
+        self.bot_plot_source = ColumnDataSource(data=dict(
+            bot_slice_x=np.repeat(0, self.resolution), bot_slice_y=np.repeat(0, self.resolution)))
+        self.right_plot_source = ColumnDataSource(data=dict(
+            left_slice_x=np.repeat(0, self.resolution), left_slice_y=np.repeat(0, self.resolution)))
+        self.source = ColumnDataSource(data=dict(
+            x=np.repeat(0, self.resolution), y=np.repeat(0, self.resolution)))
+
+        # Text input to change the distance of reach when searching for nearest data points
+        self.scatter_distance = TextInput(value="0.1", title="Scatter Distance")
+        self.scatter_distance.on_change('value', self._scatter_input)
+        self.dist_range = float(self.scatter_distance.value)
+
+        # Grouping all of the sliders and dropdowns into one column
+        sliders = [i for i in self.slider_dict.values()]
+        sliders.extend([self.x_input, self.y_input, self.output_select, self.scatter_distance])
+        self.sliders_and_selects = row(
+            column(*sliders))
+
+        # Layout creation
+        self.layout = row(self.contour_data(), self.right_plot(), self.sliders_and_selects)
+        self.layout2 = row(self.bot_plot())
+        curdoc().add_root(self.layout)
+        curdoc().add_root(self.layout2)
+        curdoc().title = 'MultiView'
+
+    def make_predictions(self, data):
+        """
+        Run the data parameter through the surrogate model which is given in prob.
+
+        Parameters
+        ----------
+        data : dict
+            Dictionary containing Ordered Dict of training points.
+
+        Returns
+        -------
+        array
+            np.stack of predicted points.
+        """
+        outputs = {i: [] for i in self.output_list}
+        print("Making Predictions")
+
+        # Parse dict into shape [n**2, number of inputs] list
+        inputs = np.empty([self.resolution**2, self.num_of_inputs])
+        for idx, values in enumerate(data.values()):
+            inputs[:, idx] = values.flatten()
+
+        # Pair data points with their respective prob name. Loop to make predictions
+        for idx, tup in enumerate(inputs):
+            for name, val in zip(data.keys(), tup):
+                self.prob[self.surrogate_comp.name + '.' + name] = val
+            self.prob.run_model()
```

Comment:
We might need to discuss some use-cases for this visualizer, but if the user gives you a prob object that has a ton of other calculations in it, then this will be needlessly slow. An alternative might be to create a new clean `Problem` instance, and add self.surrogate_comp to it, along with appropriate indepvarcomps for the inputs.  Then, when it runs, you are only running the surrogate model.